Assets/Scripts/Models/World.cs

```diff
@@ -342,6 +342,44 @@ public Utility PlaceUtility(string objectType, Tile tile, bool doRoomFloodFill =
         return PlaceUtility(util, tile, doRoomFloodFill);
     }
 
+    /// <summary>
+    /// Reserves the furniture's work spot, preventing it from being built on.
+    /// </summary>
+    /// <param name="furn">The furniture whose workspot will be reserved.</param>
+    /// <param name="tile">The tile on which the furniture is located, for furnitures which don't have a tile, such as prototypes.</param>
+    public void ReserveTileAsWorkSpot(Furniture furn, Tile tile = null)
```

Comment:
I think this should be at a Tile. The Furniture can call it as ```Jobs.GetWorkSpotTile().ReserveAsWorkSpot(this)``` and the job as ```Furniture furn = (Furniture)job.buildablePrototype; World.Current.GetTileAtOffset(job.Tile, furn.Jobs.WorkSpotOffset).ReserveAsWorkSpot(furn)```. And the HashSet in Tile can be private. But not totally sure.